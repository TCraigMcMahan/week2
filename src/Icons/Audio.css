.Audio {
    display: flex;
    flex-wrap: nowrap;
    color: gray;
    font-weight: 500;
    height: 20px;
    max-height: 20px;
    width: 224px;
    max-width: 224px;
    padding-bottom: 10px;
}

.audio-text {
    font-size: 0.75em;
    text-align: left;
    padding-right: 75px;
}

.audio-image {
    padding-right: 10px;
}

.star-container {
    height: 20px;
    width: 20px;
}

.star-container img {
    height: auto;
    width: 65%;
}
