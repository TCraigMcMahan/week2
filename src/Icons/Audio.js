import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Audio.css';
import speaker from '../img/icon-speaker.png';
import star from '../img/icon-star.svg';

class Audio extends Component {

    render() {

        if (this.props.hasAudioAvailable) {
            return (
                <div className="Audio">
                    <div className="audio-image">
                        <img src={speaker} alt="Play Audio" />
                    </div>
                    <div className="audio-text">Audio Available</div>
                    {this.props.memberPreview ? (
                        <div className="star-container">
                            <img src={star} alt="" />
                        </div>
                    ) : null}
                </div>
            )
        } else if (this.props.memberPreview) {
            return (
                <div className="Audio">
                    <div className="star-container">
                        <img src={star} alt="" />
                    </div>
                    <div className="audio-text">Member preview</div>
                </div>
            )
        } else {
            return null;
        }
    }
}


Audio.propTypes = {
    hasAudioAvailable: PropTypes.bool.isRequired,
    memberPreview: PropTypes.bool.isRequired
};


Audio.defaultProps = {
    hasAudioAvailable: false,
    memberPreview: false
};

export default Audio;