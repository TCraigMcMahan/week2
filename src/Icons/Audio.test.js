import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Audio from './Audio.js';

describe('Audio', () => {


    it('renders without crashing with data', () => {
        const div = document.createElement('div');

        ReactDOM.render(<Audio hasAudioAvailable = {true}
            memberPreview  = {true} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render with title, description, image, etc.', () => {

        const component = TestRenderer.create(<Audio hasAudioAvailable = {true}
            memberPreview  = {true}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
}); 