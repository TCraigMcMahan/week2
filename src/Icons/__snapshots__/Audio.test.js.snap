// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audio should render with title, description, image, etc. 1`] = `
<div
  className="Audio"
>
  <div
    className="audio-image"
  >
    <img
      alt="Play Audio"
      src="icon-speaker.png"
    />
  </div>
  <div
    className="audio-text"
  >
    Audio Available
  </div>
  <div
    className="star-container"
  >
    <img
      alt=""
      src="icon-star.svg"
    />
  </div>
</div>
`;
